tensorflow/python/framework/sparse_tensor.py

```diff
@@ -178,6 +178,21 @@ def values(self):
     """
     return self._values
 
+  def with_values(self, new_values):
+    """Returns a copy of `self` with `values` replaced by `new_values`.
+
+    This method produces a new `SparseTensor` that has the same nonzero
+    indices, but updated values.
+
+    Args:
+      new_values: The values of the new `SparseTensor. Needs to have the same
```

Comment:
Optional: call out the fact that the new values can have a different `dtype` than the current values

Reply:
should be done with the latest commit